unlock-wordpress-plugin/inc/classes/blocks/class-unlock-box-block.php

```diff
@@ -168,4 +179,196 @@ private function get_checkout_url( $attributes ) {
+    $post_types = get_post_types(); //detect any custom post types like courses and add lock features automatically
+    foreach ($post_types as $post_type) {
+        add_meta_box(
+            'unlock_pv2_meta_box', // ID
+            'Unlock Protocol', // title
+            array( $this, 'unlock_pv2_show_custom_meta_box' ), // callback function
+            $post_type, // post type
+            'side', // position
+            'high', // priority
+			array(
+				'locks'      => array(
+					'type'    => 'array',
+					'default' => array(),
+				),
+				'ethereumNetworks' => array(
+					'type'    => 'array',
+					'default' => array(),
+				),
+			)
+        );
+    }
 }
+
+
+
+
+
+
+// Show custom meta box with "Lock Networks" dropdown and "Lock Contract ID" input box
+ public function unlock_pv2_show_custom_meta_box() 
+{
+
+	// Add nonce for security and authentication.
+	// wp_nonce_field( 'unlock_pv2_meta_box', 'unlock_pv2_meta_box_nonce' );
+
+    global $post;
+    $meta_network = get_post_meta($post->ID, 'unlock_pv2_network_meta', true);
+    $meta_id = get_post_meta($post->ID, 'unlock_pv2_id_meta', true);
+	$networkList = get_option('unlock_protocol_settings');
+	// var_dump($networkList);
+    
+    ?>
+  
+        <button id="add-lock-button" onclick="toggleLockInputs()" class="button button-primary">Add Lock</button>
+
+        <div id="lock-network-inputs" style="display: none;">
+            <label for="unlock-pv2-network-label">Lock Networks:</label>
+            <select name="unlock-pv2-network-select" id="unlock-pv2-network-select" onchange="toggleLockIdInput()">
+                <option value="" <?php selected($meta_network, ""); ?>></option>
+                <?php foreach ($networkList['networks'] as $network) { ?>
+                    <option value="<?php echo esc_attr($network['network_name']); ?>" <?php selected($meta_network, $network['network_name']); ?>><?php echo esc_html($network['network_name']); ?></option>
+                <?php } ?>
+            </select>
+
+            <div id="lock-id-input" style="display: <?php echo ($meta_network ? "block" : "none"); ?>;">
+				<label for="unlock-pv2-lock-id-label">Lock Contract ID:</label>
+				<input type="text" name="unlock-pv2-lock-id-input" id="unlock-pv2-lock-id-input" value="<?php echo esc_attr($meta_id); ?>" />
+			</div>
+
+			<br>
+        	<div id="unlock-pv2-lock-setup-notification"></div>
+
+
+        </div>
+
+        <script>				
+
+			function toggleLockInputs() {
+                var lockInputs = document.getElementById("lock-network-inputs");
+                if (lockInputs.style.display === "none") {
+                    lockInputs.style.display = "block";
+                    document.getElementById("add-lock-button").innerHTML = "Remove Lock";
+                } else {
+                    lockInputs.style.display = "none";
+                    document.getElementById("add-lock-button").innerHTML = "Add Lock";
+                }
+            }
```

Comment:
Since this code is the same for the lock box, do you think we could just re-use?

Reply:
Will check if we can